File: 0202.快乐数/0202-快乐数.rs
use std::collections::HashSet;
impl Solution {
    pub fn is_happy(n: i32) -> bool {
        let mut seen = HashSet::new();
        let mut n = n;
        while !seen.contains(&n) {
            if n == 1 {
                return true;
            }
            seen.insert(n);
            let mut tmp = 0;
            while n > 0 {
                tmp += (n%10)*(n%10);
                n /= 10;
            }
            n = tmp;
        }
        return false;
    }
}

File: 0202.快乐数/0202-快乐数.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n not in seen:
            if n == 1:
                return True
            seen.add(n)
            tmp = 0
            while n > 0:
                tmp += (n % 10) ** 2
                n //= 10
            n = tmp
        return False
